### PR TITLE
[feat](nereids) add is_merge tag for data sink

### PR DESCRIPTION
**Pull request by englefly**
### What problem does this PR solve?
set is_merge true, if its corresponding exchange node contains sortInfo.
BE use this flag to accelerate data transferring.

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [ ] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

File: gensrc/thrift/DataSinks.thrift [hunk 1]
```diff
@@ -189,6 +189,7 @@ struct TDataStreamSink {
   11: optional i64 tablet_sink_txn_id
   12: optional Types.TTupleId tablet_sink_tuple_id
   13: optional list<Exprs.TExpr> tablet_sink_exprs
+  14: optional bool is_merge
```
Comment (morrySnow) [inline gensrc/thrift/DataSinks.thrift:192]: does `ordered` is a better name?

**Comment by englefly** [comment]
run buildall

**Review by Mryange — APPROVED** [review]

File: gensrc/thrift/DataSinks.thrift (hunk 1 above)
Comment (englefly) [reply gensrc/thrift/DataSinks.thrift:192]: The intuition behind this is sink node must receive reply message from exchange before starting a new data block transfer. "order" is one of the most common senario, but not all.

**Review by starocean999 — APPROVED** [review]

**Comment by englefly** [comment]
it is better to determine this by FE than that by BE.
for example, if data spill to disk, the node before sink is scan, not sort

**Pull request merged**